### ISSUE TITLE
Issue building DAE with CMAKE

### ISSUE THREAD

**Issue by sKamer73** [dae-cpp/dae-cpp#31]
Hello!

I came accross your work and found it very interesting for what I'd like to do: simulate a 1D thermocline system.
I've been trying, and it's been failing for the past 5 hours. I'm not sure that I understand the issue.
When building the .sln file with Cmake-Gui, I come accross the issue that different libraries are not found for building test projects: LIB_DL,LIB_MATH, LIB_MKL_GNU_THREAD and LIB_PTHREAD. On the other hand, LIB_MKL_CORE and LIB_MKL_INTEL_LP64 was found (after I changed the correct path to MKL folder).

I'm not sure that the issue is from DAE's Cmake, but maybe from Intel configuration: the basic exemple Vector_Add doesn't run, raising the exception "OpenCL API failed. OpenCL API returns: -999 (Unknown OpenCL error code) -999 (Unknown OpenCL error code)"

Would you have any hints?

Thanks!

**Comment by sKamer73** [comment]
Re-Hi,

I'm thinking about giving further details. I worked out the Vector_Add issue, by running it in the Intel oneApi command prompt, and setting SYCL_DEVICE_TYPE = CPU, although I don't know what it means. I followed those instructions: [here](https://community.intel.com/t5/Intel-oneAPI-Base-Toolkit/OneAPI-sampes-are-throwing-an-exception-in-VisualStudio-2019-16/td-p/1281689)

Further precisions: 
- CMAKE image
![image](https://user-images.githubusercontent.com/57094598/122562582-86547400-d043-11eb-98ae-78906940430f.png)
- lib folder of mkl -> which library are which? If that is the issue..
![image](https://user-images.githubusercontent.com/57094598/122562753-b4d24f00-d043-11eb-8346-3266037b5a76.png)

Thanks!

**Comment by ikorotkin** [comment]
Hi @sKamer73,

Thanks for your detailed report. I have just tried to reproduce these errors using similar software, `cmake` for Windows, Visual Studio, and for me it works straight out of the box. Cmake found all libraries automatically, Visual Studio produced all exe-files with one click. But in your case it cannot find some basic libraries.

Do you have Visual Studio installed with the C++ compiler? I am asking because technically Visual Studio can be installed without C++ compiler and libraries. Can you open it and create and compile a simple 'hello world' program?

When you run `cmake` it asks you to choose your compiler, for example:

![cmake_win](https://user-images.githubusercontent.com/29599800/122614311-524a7480-d07e-11eb-896b-4e6fd55677ff.png)

It should correspond to the compiler you have installed.

And I see you use OneAPI Intel MKL 2021, the most recent version. Intel reworked MKL library quite a lot starting from 2021. It still works under Linux (if you provide correct paths) but I didn't try it under Windows. Can you try to roll back to Intel MKL 2019?

**Comment by sKamer73** [comment]
Hi!

Quite excited as I managed to run DAE! Issue was from the different link to libraries, that was not done as CMake would not find the different libraries names. By playing at "build & find the next missing library", I was able to run diffusion_2d, as per the photo below:

![image](https://user-images.githubusercontent.com/57094598/122754994-a29b1f80-d294-11eb-93df-23aff10ee3b7.png)

To summarize the process followed:
-configuration and generation with CMAke, even though error appeared regarding libraries not found
-opening the solution file .sln and added the following to each project (even dae_static):
1. Link Editor> General >Repertory of supplementary library: C:\Program Files (x86)\Intel\oneAPI\mkl\latest\lib\intel64
2. Link Editor > Entry > Supplementary dependency: mkl_sequential.lib;mkl_intel_thread.lib;mkl_core.lib;mkl_intel_ilp64.lib
3. C++ > General > Other Include repertory: C:\Program Files (x86)\Intel\oneAPI\mkl\2021.2.0\include
For daecpp_static, as it only generates a .lib, those options were entered in the Librarian > General.

Sorry for translations from french to english, probably not the best. I had also tried to find MKL2019, but quite hard to get - I only found a python installation, which might have been enough (and no login, etc., necessary, as well as much smaller library than oneAPI). But it didn't work out (missing dll, didn't go further on it).

Thanks!
